async_asgi_testclient/tests/asgi_spec/test_http.py

```diff
@@ -0,0 +1,391 @@
+    that corresponds to the ASGI version the server implements.
+    https://asgi.readthedocs.io/en/latest/specs/main.html#applications
+
+    We expect this to be version 3.0, as that is the current spec version.
+    """
+
+    async def handle_all(scope, receive, send):
+        if scope["type"] == "http":
+            assert "asgi" in scope
+            assert scope["asgi"]["version"] == "3.0"
+
+    mock_app.handle_all = handle_all
+
+    async with TestClient(mock_app) as client:
+        await client.get("/")
+
+
+@pytest.mark.asyncio
+async def test_http_spec_version_is_missing_or_correct(mock_app):
+    """
+    asgi["spec_version"] (Unicode string) – Version of the ASGI HTTP spec this server
+     understands; one of "2.0" or "2.1". Optional; if missing assume 2.0
+    https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope
+
+    TestClient doesn't specify spec_version at the moment, which is also okay.
+    Note that if newer features are added (specifically allowing None in 'server' scope value)
+    then the spec_version needs to be passed correctly.
+    """
+
+    async def handle_all(scope, receive, send):
+        if scope["type"] == "http":
+            assert (
+                "spec_version" not in scope["asgi"]
+                or scope["asgi"]["spec_version"] == "2.0"
+            )
+
+    mock_app.handle_all = handle_all
+
+    async with TestClient(mock_app) as client:
+        resp = await client.get("/")
+        assert resp.status_code == 200
+
+
+@pytest.mark.asyncio
+async def test_http_version_is_1_1(mock_app):
+    """
+    http_version (Unicode string) – One of "1.0", "1.1" or "2".
+    https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope
+    TestClient currently only supports http/1.1, so test that it is set correctly.
+    """
+
+    async def handle_all(scope, receive, send):
+        if scope["type"] == "http":
+            assert scope["http_version"] == "1.1"
+
+    mock_app.handle_all = handle_all
+
+    async with TestClient(mock_app) as client:
+        resp = await client.get("/")
+        assert resp.status_code == 200
+
+
+@pytest.mark.xfail(
+    AssertionError, reason="TestClient does not uppercase scope['method']"
+)
+@pytest.mark.asyncio
+async def test_http_method_is_uppercased(mock_app):
+    """
+    method (Unicode string) – The HTTP method name, uppercased.
+    https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope
+    Method is uppercased.
+    """
+
+    async def handle_all(scope, receive, send):
+        if scope["type"] == "http":
+            assert scope["method"] == "GET"
+
+    mock_app.handle_all = handle_all
+
+    async with TestClient(mock_app) as client:
+        resp = await client.open("/", method="get")
+        assert resp.status_code == 200
+
+
+# Ok, this one I'm not sure about...
+# The grey area here is TestClient doesn't currently decode an encoded URL; but should it?
```

Comment:
I think the TestClient should encode the `path` internally, the same way we are doing for the `query_string`:

https://github.com/vinissimus/async-asgi-testclient/blob/56779ef69ff2eebd88178980d6f8ba50b91b1529/async_asgi_testclient/utils.py#L49-L53

If you want to change it and add/modify one of  the integration tests to ensure the frameworks still work I think it's enough :)